vrx_ign/src/PolyhedronVolume.hh

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (C) 2019 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#ifndef VRX_POLYHEDRONVOLUME_HH_
+#define VRX_POLYHEDRONVOLUME_HH_
+
+#include <vector>
+#include <ignition/math/Vector3.hh>
+#include <ignition/math/Quaternion.hh>
+
+namespace vrx
+{
+  /// \brief Represents a plane as a normal and offset.
+  struct Plane
+  {
+    /// \brief Initializes plane at z=0.
+    Plane();
+
+    /// \brief Vector3 normal to plane.
+    ignition::math::Vector3d normal;
+
+    /// \brief Offset w.r.t. normal.
+    float offset;
+  };
+
+  /// \brief Represents output volume with centroid.
+  struct Volume
+  {
+    Volume();
+
+    /// \brief Overloads += for volume object.
+    Volume &operator+=(const Volume &_rhs);
+
+    /// \brief Submerged volume of shape.
+    double volume;
+
+    /// \brief Vector3 representing volume centroid.
+    ignition::math::Vector3d centroid;
+  };
+
+  /// \brief Submerged volume calculation using polyhedron
+  /// based on: Exact Buoyancy for Polyhedra by Eric Catto.
```

Comment:
Recommend a complete citation: 
Erin Catto. “Exact Buoyancy for Polyhedra”. In Game Programming Gems 6, Charles River Media, 2006, pp. 175–187.

Reply:
Modified in [19d536c](https://github.com/osrf/vrx/pull/500/commits/19d536c20c950b825ac3c5bb0ee970eebddbfdce).